src/ray/gcs/redis_context.cc

```diff
@@ -21,20 +21,20 @@ namespace {
 /// A helper function to call the callback and delete it from the callback
 /// manager if necessary.
 void ProcessCallback(int64_t callback_index,
-                     const ray::gcs::CallbackReply &callback_reply) {
+                     std::shared_ptr<ray::gcs::CallbackReply> callback_reply) {
```

Comment:
How about use `unique pointer` instead of `shared pointer` ? As we already know the owner of CallbackReply is the callback and no one else needs this object.

Reply:
Ah yeah unfortunately I learned that you can't `std::move` a `unique_ptr` into a lambda, so we're forced to write a really ugly workaround or use a `shared_ptr`

Reply:
Maybe we can use reference of unique_ptr to resolve this problem.

Reply:
I don't think so. That was a bug in the original code - if you pass a
unique_ptr reference into the lambda it doesn't keep the object alive, so
you can easily get segfaults when running the callback as the actual
unique_ptr goes out of scope after placing the callback on the event loop.